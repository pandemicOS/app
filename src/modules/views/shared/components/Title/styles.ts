import { StyleSheet } from 'react-native';

import { STYLE } from '../../../../../constants';

export default StyleSheet.create({
  title: {
    color: STYLE.COLOR.BLACK,
    fontSize: STYLE.FONT.SIZE.HEADING_BIG,
    fontWeight: STYLE.FONT.WEIGHTS.MEDIUM,
  }
});
