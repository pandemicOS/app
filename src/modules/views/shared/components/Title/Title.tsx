import React from 'react';
import { Text, StyleProp, TextStyle } from 'react-native';

import styles from './styles';

export interface ITitleProps {
  children: string | string[];
  style?: StyleProp<TextStyle>;
}

export default class Title extends React.PureComponent<ITitleProps> {
  public render() {
    const { children, style } = this.props;
    return (
      <Text style={[styles.title, style]}>{children}</Text>
    );
  }
}
