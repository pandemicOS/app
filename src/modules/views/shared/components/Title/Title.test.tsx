import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Title, { ITitleProps } from './Title';

describe('Title', () => {
  let props: ITitleProps;
  beforeEach(() => {
    props = {
      children: 'Title'
    };
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(<Title {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
