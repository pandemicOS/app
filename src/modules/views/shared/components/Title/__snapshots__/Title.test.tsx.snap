// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders without crashing 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1E272E",
        "fontSize": 25,
        "fontWeight": "600",
      },
      undefined,
    ]
  }
>
  Title
</Text>
`;
